### DISCUSSION TITLE
Adding settings to specialized image

### DISCUSSION THREAD

**Discussion by dakky** [arendst/Tasmota#18940] [Support request] [answered]
Hello ;-)

I want to bring tasmota to my Shelly plus 2pm devices. For these there is already the esp32solo1 image. I want to add settings like wifi, mqtt and so to the image. so i need to rebuild them. 

Now my issue: where do I get the "base" configuration for the esp32solo1 image, so i can add my overrides?

**Comment by barbudor** [comment, marked as answer]
All build shares the same setting file
You put all your custom settings in user_config_override.h and then you build the standard solo1 image and it will include your changes

**Reply by barbudor** [reply]
Note that it's simple like that for **settings**

If you want to add features it may be more complex so better be explicit on what you want to add

**Reply by dakky** [reply]
can you redirect me to the correct location where this image is build. I thougt every image is the default tasmota repo with a "special" config.h, which defines, which features are included. seems I am wrong ;)

**Reply by barbudor** [reply]
All esp32 targets are defined in platformio_tasmota_env32.ini
For solo1:
```
[env:tasmota32solo1]
extends                 = env:tasmota32_base
board                   = esp32_solo1
build_flags             = ${env:tasmota32_base.build_flags}
                          -DFIRMWARE_TASMOTA32
                          -DCODE_IMAGE_STR='"solo1"'
                          -DOTA_URL='"http://ota.tasmota.com/tasmota32/release/tasmota32solo1.bin"'
```
So solo1 image is using all default features from FIRMWARE_TASMOTA32 which are defined in tasmota_configurations_ESP32.h

But you shouldn't modify any of Tasmota standard files. Everything is done so you can override settings easily in user_config_override.h without modifying any file that are under git control. This allows to blindly update your locally clone git and rebuild without any headache.

In any doubt, read again https://tasmota.github.io/docs/Compile-your-build/#customize-your-build

**Reply by dakky** [reply]
i succeded to build and customize the image. thanks for your guidance ;) the relevant hint was the name of the pio env

```
podman run -ti --rm -v $(pwd)/Tasmota:/tasmota  docker-tasmota -e tasmota32solo1 
```
